packages/react-aria-components/src/Button.tsx

```diff
@@ -161,18 +161,16 @@ export const Button = /*#__PURE__*/ createHideableComponent(function Button(prop
   );
 });
 
-function disablePendingProps(props) {
+function useDisableInteractions(props, isPending) {
   // Don't allow interaction while isPending is true
-  if (props.isPending) {
-    props.onPress = undefined;
-    props.onPressStart = undefined;
-    props.onPressEnd = undefined;
-    props.onPressChange = undefined;
-    props.onPressUp = undefined;
-    props.onKeyDown = undefined;
-    props.onKeyUp = undefined;
-    props.onClick = undefined;
+  if (isPending) {
+    for (const key in props) {
+      if (key.startsWith('on')) {
+        props[key] = undefined;
+      }
```

Comment:
what about `onBlur`/`onFocus`? Should those be removed?

Reply:
one question i did have here, is should i move this logic into useButton, it might be easier, but currently useButton doesn't know about pending since there are so many style dependent and dom structure requirements, so it only exists in higher levels

Reply:
hm, is this mainly a concern about how the spinner may be rendered relative to the button and/or if there is a spinner at all? Not quite sure I fully understand the style/dom structure dependencies here

Reply:
essentially there are different ways you can set things up
https://react-spectrum.adobe.com/react-aria/Button.html#accessibility